src/nimib/blocks.nim

```diff
@@ -16,15 +16,16 @@ func nbNormalize*(text: string): string =
   text.replace("\c\l", "\n").replace("\c", "\n").strip # this could be made more efficient
 # note that: '\c' == '\r' and '\l' == '\n'
 
-template newNbBlock*(cmd: string, nbDoc, nbBlock, body, blockImpl: untyped) =
+template newNbBlock*(cmd: string, nbDoc, nbBlock, readCode, body, blockImpl: untyped) =
```

Comment:
`readCode` should be bool and we could position it after `cmd: string`. We probably could also add the same old api where readCode is defaulted to true (not sure if template composition runs into issues there...)

Reply:
Moving it to the front sounds good 👍 I'm a bit afraid of adding types to templates now after battling with them yesterday. (saw that you had also come across [this](https://github.com/nim-lang/Nim/issues/16680)) but as you already have a `string` there it should be fine. Do you mean to default it as a `readCode: bool = true` or making a separate overload without it that just calls this new one? That's also something I battled with yesterday as optional parameters aren't supported in templates with untyped arguments without the `experimental` pragma. So the separate overload is really the only option. (if template composition behaves as we want it to as you said 🤣)

Reply:
Yep separate overload. I also had issues with optional arguments in templates.

Reply:
I'll give it a try after lunch. Feels good to have a test suite in nimib to rely on with these things now :)

Reply:
Yes that is another Boone of recent PR... next step should be automatic generation of documentation with netlify/vercel preview as in scinim/getting-started

Reply:
Yes please

Reply:
Yes please to which one of them? 😅 Old or new?